docs/sources/developer/architecture/components/api.md

```diff
@@ -0,0 +1,78 @@
+# API
+
+The Faro API is the primary interface for end-users to interact with Faro. It is the central place where signals are
+sent to but it also contains helpers methods for other components like metas.
+
+## Signals
+
+### Events
+
+Events are signals related to tracking various actions like mouse clicks as well as other events like transitioning to
+another page.
+
+Methods:
+
+- `pushEvent` - sends an event
+
+### Exceptions
+
+Exceptions are signals that are represented by errors which occurred in an app.
+
+When an exception is sent, the API will pass it through a stacktrace parser to extract the relevant information from it.
+By default, the core package does not contain any stacktrace parsers and they are either provided by wrapper packages
+like `web-sdk` or by the user.
+
+Methods:
+
+- `changeStacktraceParser` - replaces the existing stacktrace parser with a new one
+- `getStacktraceParser` - returns the current stacktrace parser
+- `pushError` - sends an exception
+
+### Logs
+
+Logs are signals which capture events that happen within the app itself. They do not have a business meaning but a
+technical meaning.
```

Comment:
nit: maybe "technical meaning" could be replaced with "have use for operators as a means of troubleshooting or observing historic (or current) application behaviour"